### PR TITLE
Avoid special casing work items in `sendtohelixhelp.proj`

### PR DESCRIPTION
**Pull request by MaximLipnin**
Addresses https://github.com/dotnet/runtime/issues/46156

### CODE REVIEW

**Comment by MaximLipnin** [comment]
[Wasm Console Sample helix log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-46315-merge-9f3b816ce4a3471396/WasmConsoleSample/console.d38e5758.log?sv=2019-07-07&se=2021-01-12T09%3A33%3A43Z&sr=c&sp=rl&sig=YnIK87CAoz8vg3N8gQ9wwAj0u7rEhx62%2B5j66%2Bkos5I%3D)

[Wasm Browser Sample helix log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-46315-merge-39c34435cc2a44ceac/WasmBrowserSample/console.eb308c99.log?sv=2019-07-07&se=2021-01-12T09%3A33%3A43Z&sr=c&sp=rl&sig=3niW3uW83O4ygPI05OlyV%2BvAlgKfcIgEfM5NE0yD0hk%3D)

File: src/mono/netcore/sample/wasm/browser/WasmSample.csproj [hunk 1]
```diff
@@ -60,6 +60,19 @@
...
   </ItemGroup>
 
+  <Target Name="PrepareRunScript" BeforeTargets="CopySampleAppToHelixTestDir" Condition="'$(ArchiveTests)' == 'true'" >
+    <PropertyGroup>
+      <RunScriptCommand>
+        $HARNESS_RUNNER wasm $XHARNESS_COMMAND  --app=. --browser=Chrome  --html-file=index.html --output-directory=$HELIX_WORKITEM_UPLOAD_ROOT/xharness-output -- WasmSample.dll --testing
+      </RunScriptCommand>
+    </PropertyGroup>
+
+    <MSBuild Projects ="$(RepoRoot)eng/testing/tests.targets"
```
Comment (safern) [inline src/mono/netcore/sample/wasm/browser/WasmSample.csproj:70]:
```suggestion
    <MSBuild Projects="$(RepoRoot)eng/testing/tests.targets"
```

File: src/mono/netcore/sample/wasm/console/WasmSample.csproj [hunk 2]
```diff
@@ -53,6 +54,19 @@
...
   </ItemGroup>
 
+  <Target Name="PrepareRunScript" BeforeTargets="CopySampleAppToHelixTestDir" Condition="'$(ArchiveTests)' == 'true'" >
+    <PropertyGroup>
+      <RunScriptCommand>
+        $HARNESS_RUNNER wasm $XHARNESS_COMMAND  --app=. --engine=$(JSEngine) $(JSEngineArgs) --js-file=runtime.js --output-directory=$XHARNESS_OUT --  $(RunTestsJSArguments) --run $(AssemblyName).dll
+      </RunScriptCommand>
+    </PropertyGroup>
+
+    <MSBuild Projects ="$(RepoRoot)eng/testing/tests.targets"
```
Comment (safern) [inline src/mono/netcore/sample/wasm/console/WasmSample.csproj:64]: Why not import the `tests.targets` file and just set your target to depend on `GenerateRunScript`? Would that work?

Comment (MaximLipnin) [reply src/mono/netcore/sample/wasm/console/WasmSample.csproj:64]: If the "tests.targets" file is imported into the project, there will be such error during build: 
`error MSB4011: "/runtime/src/mono/wasm/build/WasmApp.targets" cannot be imported again. It was already imported at "/runtime/src/mono/netcore/sample/wasm/console/WasmSample.csproj (73,3)". This is most likely a build authoring error. This subsequent import will be ignored.`

Comment (safern) [reply src/mono/netcore/sample/wasm/console/WasmSample.csproj:64]: I see... what you could do is remove the `WasmApp.targets` import in `WasmSample.csproj` and then that issue would be gone. 

https://github.com/dotnet/runtime/blob/e81753c6542d1a45d6bab0020837053c5e3f05cc/src/mono/netcore/sample/wasm/console/WasmSample.csproj#L87

File: src/mono/netcore/sample/wasm/browser/WasmSample.csproj [hunk 3]
```diff
@@ -60,6 +60,19 @@
     </Content>
   </ItemGroup>
 
+  <Target Name="PrepareRunScript" BeforeTargets="CopySampleAppToHelixTestDir" Condition="'$(ArchiveTests)' == 'true'" >
```
Comment (safern) [inline src/mono/netcore/sample/wasm/browser/WasmSample.csproj:63]: We can do the same as we did for the console sample, right?

Comment (steveisok) [reply src/mono/netcore/sample/wasm/browser/WasmSample.csproj:63]: Yeah, I'll push that one up too.

**Comment by MaximLipnin** [comment]
After the merge with the latest master there is another error
```
eng/illink.targets(215,11): error MSB4057: (NETCORE_ENGINEERING_TELEMETRY=Build) The target "GetBinPlaceTargetFramework" does not exist in the project.  /__w/1/s/eng/illink.targets
```

**Comment by safern** [comment]
It seems like that was caused because `tests.mobile.targets` now import `illink.targets`. https://github.com/dotnet/runtime/commit/1e6826a8ff8345d3ec97ac62e900f4086b7de6c1#diff-56a2c52766d355ebc47b5efbd7bedb042780d0945ba5353a315fc162ed988122R224

What we could do is condition that import to a property like `MobileTestsSkipLinkImport != true` and then set that property to true on the sample apps.

**Comment by akoeplinger** [comment]
@MaximLipnin would you mind updating this PR and solving the merge conflicts?

**Comment by ViktorHofer** [comment]
gentle ping @MaximLipnin

**Comment by MaximLipnin** [comment]
It will be addressed in other PR, closing it

**Pull request closed without merging**